nlp/optimize/tron.py

```diff
@@ -319,13 +315,17 @@ def projected_linesearch(self, x, l, u, g, d, H, alpha=1.0):
 
         where H=Hᵀ and g is a vector. Given the parameter α, the step is
 
-           s[α] = P[x + α d] - x,
+           s[α] = s₀ + P[x + α d] - x,
```

Comment:
I don't recommend doing this because if s0 is already close to a solution, additional steps are likely to be small. Then you're adding relatively small quantities (the successive steps) to a relatively large quantity (s0) and run the risk of severe cancellation. It's a better idea numerically to first add the steps together so they're not too small, and add the final result to s0. That's what the code currently does.